lib/identity/hostdata.rb

```diff
@@ -15,7 +15,7 @@ class MissingConfigError < StandardError; end
     # @return [String]
     def self.domain
       @domain ||= begin
-        File.read(File.join(root.to_s, DOMAIN_PATH)).chomp
+        ENV['LOGIN_DOMAIN'] || File.read(File.join(root.to_s, DOMAIN_PATH)).chomp
```

Comment:
when an env var is not there, the value will be nil. So it `LOGIN_DOMAIN` was nil, we'd be hitting both ENV + disk every time... I think we need to an approach more like:

```
return @domain if defined?(@domain)

@domain = # blah blah blah
```

For all of these. It looks like a plain has read but `ENV#[]` is slow IIRC so I would definitely like to cache this even if it's nil

Reply:
Good catch, will do